stubs/entrypoints/entrypoints.pyi

```diff
@@ -0,0 +1,59 @@
+import sys
+from typing import Any, Dict, Generic, Iterator, List, Sequence, Text, Tuple, Type, TypeVar
+
+if sys.version_info >= (3, 0):
+    from configparser import ConfigParser
+else:
+    from ConfigParser import ConfigParser
+
+if sys.version_info >= (3, 8):
+    from re import Pattern
+else:
+    from typing import Pattern
+
+_T = TypeVar("_T")
+
+entry_point_pattern: Pattern[Text]
+file_in_zip_pattern: Pattern[Text]
+
+class BadEntryPoint(Exception):
+    epstr: Text
+    def __init__(self, epstr: Text) -> None: ...
+    @staticmethod
+    def err_to_warnings() -> Iterator[None]: ...
+
+class NoSuchEntryPoint(Exception):
+    group: Text
+    name: Text
+    def __init__(self, group: Text, name: Text) -> None: ...
+
+class EntryPoint(Generic[_T]):
+    _E = TypeVar("_E", bound="EntryPoint[_T]")
```

Comment:
Please advise - is this not allowed in general or just an issue with typecheckers other than `mypy`? Saw this approach described in [Precise typing of alternative constructors](https://mypy.readthedocs.io/en/stable/more_types.html#precise-typing-of-alternative-constructors) first, using it quite extensively since then.